applications/DamApplication/custom_utilities/construction_utility.hpp

```diff
@@ -0,0 +1,723 @@
+
+    double time = mrThermalModelPart.GetProcessInfo()[TIME];
+    time = time/mTimeUnitConverter;
+    double ambient_temp = mrTableAmbientTemp.GetValue(time-1);
+
+    if(size != 0)
+    {
+        for(unsigned int j = 0; j<size; ++j)
+        {
+            for(unsigned int i = 0; i<nnodes; ++i) 
+            {
+                ModelPart::NodesContainerType::iterator it_thermal = it_begin_thermal + i;
+
+                if (it_thermal->Id() == ConditionNodeIds[j]) 
+                {
+                    const double temp_current = it_thermal->FastGetSolutionStepValue(TEMPERATURE);
+                    const double heat_flux = mH0*(ambient_temp - temp_current);
+                    it_thermal->FastGetSolutionStepValue(FACE_HEAT_FLUX) = heat_flux; 
+                }
+            }
+        }    
+    }
+
+    KRATOS_CATCH("");
+}
+
+//----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+void DeactiveFaceHeatFluxStep(std::vector<IndexType> ConditionNodeIds)
+{
+    KRATOS_TRY;
+
+    const unsigned int size = ConditionNodeIds.size();
+    const unsigned int nnodes = mrThermalModelPart.GetMesh(mMeshId).Nodes().size();
+    ModelPart::NodesContainerType::iterator it_begin_thermal = mrThermalModelPart.GetMesh(mMeshId).NodesBegin();
+
+    if(size != 0)
+    {
+        for(unsigned int j = 0; j<size; ++j)
+        {
+            for(unsigned int i = 0; i<nnodes; ++i) 
+            {
+                ModelPart::NodesContainerType::iterator it_thermal = it_begin_thermal + i;
+
+                if (it_thermal->Id() == ConditionNodeIds[j]) 
+                {   
+                    //Setting to 0 the fluxes since in the next step 
+                    it_thermal->FastGetSolutionStepValue(FACE_HEAT_FLUX) = 0.0; 
+                }
+            }
+        }    
+    }
+
+    KRATOS_CATCH("");
+}
+
+///----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+protected:
+
+    /// Member Variables
+
+    ModelPart& mrMechanicalModelPart;
+    ModelPart& mrThermalModelPart;
+    int mNumNode;
+    std::size_t mMeshId;
+    std::string mGravityDirection;
+    std::string mMechanicalSoilPart;
+    std::string mThermalSoilPart;
+    std::string mSourceType;    
+    double mReferenceCoordinate;
+    double mHeight;
+    int mPhases;
+    double mH0;
+    double mTimeUnitConverter;    
+    double mAlphaInitial;
+    bool mAging;
+    double mYoungInf;
+    unsigned int mMechanicalLastCondition;
+    unsigned int mThermalLastCondition;
+    TableType& mrTableAmbientTemp;
+    
+//----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
```

Comment:
I would move the particular methods of the utility from public to protected (ActiveHeatFluxAzenhaAging...)